### PR TITLE
IE11 Modifications

### PR DESCRIPTION
**Pull request by keyurshah**
This PR contains fixes needed to allow Alpine functionality in IE11 addressed in #72 

These changes were made in collaboration with @SimoTod 

Modifications include
* modified rollup system that allows a separate build to be created specifically for IE11
* modified rollup config also allows for IE11 specific code to be removed at compile time for the standard build
* added `$refs`, `$event`, `$el` properties to objects before proxies are created.  A limitation of IE Proxies is that new properties cannot be added once a proxy is created
* elements marked with `x-ref` are added as properties before proxies are created
* added a check to adjust the handling of `CustomEvent`
* added new polyfills for missing browser functionality
* an example file specific to IE11 to test out features

Thanks!

### CODE REVIEW

**Comment by calebporzio** [comment]
Thanks for this, can we make it so that the IE11 bundle includes all the polyfills and anything Alpine needs to work in IE11?

Also, is it possible to have 2 separate rollup config files? If not, not a big deal

Thanks again for this!

**Comment by keyurshah** [comment]
sure, will look to see if I can separate the configs and bundle polyfills

**Comment by keyurshah** [comment]
was able to separate the rollup config into 2 separate files. had to install the concurrently package so watch could run on both of the separate config files.

**Comment by keyurshah** [comment]
Hey Caleb,

Thanks to @SimoTod the requested changes are almost done.

I can create 2 output files for ie11, one with and one without polyfills? I can see situations where it would be beneficial to not have polyfills included.

We can only mention the polyfills included version in the documentation to keep it simple.

Please let me know how you would prefer it.

Thanks!

File: README.md [hunk 1]
```diff
@@ -29,13 +29,17 @@ Include it in your script.
 import 'alpinejs'
 ```
 
-For IE11, polyfills will need to be provided. Please load the following scripts before the Alpine script above.
+For IE11, a different Alpine script is used and polyfills need to be provided. Please add the following scripts to the end of your `<head>`> section.
 ```html
-<script src="https://polyfill.io/v3/polyfill.min.js?features=MutationObserver%2CArray.from%2CArray.prototype.forEach%2CMap%2CSet%2CArray.prototype.includes%2CString.prototype.includes%2CPromise%2CNodeList.prototype.forEach%2CObject.values%2CReflect%2CReflect.set"></script>
-
+<script src="https://polyfill.io/v3/polyfill.js?features=MutationObserver%2CArray.from%2CArray.prototype.forEach%2CMap%2CSet%2CArray.prototype.includes%2CString.prototype.includes%2CPromise%2CNodeList.prototype.forEach%2CObject.values%2CReflect%2CReflect.set%2CString.prototype.startsWith%2CArray.prototype.find%2CArray.prototype.findIndex%2CElement.prototype.closest%2CElement.prototype.remove%2CCustomEvent%2CElement.prototype.classList%2CHTMLTemplateElement"></script>
```
Comment (philippbosch) [inline README.md]: I believe we should keep using the minified version here (`.min.js`)? Same in _examples/index_ie11.html_

Comment (keyurshah) [reply README.md]: thanks for the input! with the new polyfills included structure, polyfill.io is removed entirely.

**Comment by keyurshah** [comment]
There wasn't much difference in file size between a polyfill included and not included version. So I just simplified it and only have the polyfills included version. 

Thanks!

**Comment by timrspratt** [comment]
@keyurshah / @SimoTod  nice work! Is `dist/alpine-ie11.js` and the `.map` needed? Or have they been left in there from a name change to underscore?

**Comment by keyurshah** [comment]
@timrspratt thanks, just removed them

**Comment by calebporzio** [comment]
Thanks for all the hard work on this @keyurshah!

**Comment by philippbosch** [comment]
Thank you so much, @keyurshah!!

**Pull request merged**